### PR TITLE
Documents scopes in Microsoft Graph Toolkit components

### PR DESCRIPTION
**Pull request by waldekmastykarz**
Documents scopes in Microsoft Graph Toolkit components. Closes https://github.com/microsoftgraph/microsoft-graph-toolkit/issues/720

### CODE REVIEW

**Comment by waldekmastykarz** [comment]
@nmetulev, @elisenyang could you please review the PR? 🙂

File: concepts/toolkit/components/people-picker.md [hunk 1]
```diff
@@ -132,13 +132,13 @@ The following examples shows how to use the `error` template.
...
-| API                                                                                                              | Permission  |
-| ---------------------------------------------------------------------------------------------------------------- | ----------- |
-| [/me/people](/graph/api/user-list-people)                    | People.Read        |
-| [/users](/graph/api/user-list)  | User.ReadBasic.All |
-| [/groups](/group-list)  | Group.Read.All |
-| [/groups/\${groupId}/members](/graph/api/group-list-members) | GroupMember.Read.All        |
-| [/users/${userPrincipleName} ](/graph/api/user-get)  | User.Read |
+| Configuration | Permission | API
+| --- | ---------- | ------- |
+| `group-id` set | People.Read, User.Read.All | [/groups/\${groupId}/members](/graph/api/group-list-members) |
```
Comment (beth-panx) [inline concepts/toolkit/components/people-picker.md:137]: Seems like this should be Group.Read.All based on [this function](https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/87be56d08cfd3688c08a8c13841086e060f4899e/packages/mgt-components/src/graph/graph.groups.ts#L249) which matches the docs link.
```suggestion
| `group-id` set | Group.Read.All | [/groups/\${groupId}/members](/graph/api/group-list-members) |
```

File: concepts/toolkit/components/people-picker.md [hunk 2]
```diff
@@ -132,13 +132,13 @@ The following examples shows how to use the `error` template.
...
+| `type` set to `Person` or `any` | People.Read | [/me/people](/graph/api/user-list-people) |
+| `type` set to `Group` or searching for users and `type` set to `Group` or `any` | Group.Read.All | [/groups](/group-list) |
```
Comment (beth-panx) [inline concepts/toolkit/components/people-picker.md:139]: broken link

File: concepts/toolkit/components/people.md [hunk 3]
```diff
@@ -88,9 +87,14 @@ The following examples shows how to use the `person` template.
...
+| Configuration | Permission | API
+| --- | ---------- | ------- |
+| `groupId` set | User.Read.All, People.Read | [/groups/\${groupId}/members](/graph/api/group-list-members) |
+| `userIds` set | User.ReadBasic.All | [/users/${userId}](/graph/api/user-get) |
+| `peopleQueries` set | People.Read | [/me/people](/graph/api/user-list-people) |
+| `resource` set | Permissions specified in `scopes` | Specified in `resource` |
+| default configuration | People.Read | [/me/people](/graph/api/user-list-people) |
+| `showPresence` set | Presence.Read.All | [/communications/getPresencesByUserId](/graph/api/cloudcommunications-getpresencesbyuserid) |
 
 When using the default templates, additional APIs and permissions are required. The default template for this component uses a [mgt-person](person.md) component, which requires the following.
```
Comment (beth-panx) [inline concepts/toolkit/components/people.md:100]: Remove mgt-person permissions for consistency of change.

File: concepts/toolkit/components/person.md [hunk 4]
```diff
@@ -204,15 +204,20 @@ The following properties are available on the config object.
...
-| [/me](/graph/api/user-get)                              | User.Read          |
-| [/me/photo/$value](/graph/api/profilephoto-get)        | User.Read          |
-| [/me/people/?$search=](/graph/api/user-list-people)     | People.Read        |
-| [/me/contacts/\*](/graph/api/user-list-contacts&tabs=cs) | Contacts.Read      |
-| [/users/{id}/photo/$value](/graph/api/user-list-people) | User.ReadBasic.All |
-| [/me/presence](/graph/api/presence-get)                | Presence.Read |
-| [/users/{id}/presence](/graph/api/presence-get)        | Presence.Read.All |
+| Configuration | Permission | API |
+| ------------- | ---------- | --- |
+| `personDetails` set without image, `fetchImage` set to `true`, `avatarType` set to `photo`, retrieved person is a contact and `useContactApis` set to `true` | Contacts.Read | [/me/contacts/\*](/graph/api/user-list-contacts&tabs=cs) |
```
Comment (beth-panx) [inline concepts/toolkit/components/person.md:209]: broken link

File: concepts/toolkit/components/person.md [hunk 5]
```diff
@@ -204,15 +204,20 @@ The following properties are available on the config object.
...
+| `personDetails` set without image, `fetchImage` set to `true`, `avatarType` set to `photo` and person is not a contact or `useContactApis` is set to `false` | User.ReadBasic.All | [/users/{id}/photo/$value](/graph/api/user-list-people) |
```
Comment (beth-panx) [inline concepts/toolkit/components/person.md:210]: Wrong link to get photo. Should it be this one /graph/api/profilephoto-get ?

File: concepts/toolkit/components/person.md [hunk 6]
```diff
@@ -204,15 +204,20 @@ The following properties are available on the config object.
...
+| `personDetails` set without image, `fetchImage` set to `true`, `avatarType` set to `photo` and user specified via email | User.ReadBasic.All | [/users/{id}/photo/$value](/graph/api/user-list-people) |
```
Comment (beth-panx) [inline concepts/toolkit/components/person.md:211]: Wrong link to get photo. Should it be this one /graph/api/profilephoto-get ?

File: concepts/toolkit/components/person.md [hunk 7]
```diff
@@ -204,15 +204,20 @@ The following properties are available on the config object.
...
+| `personDetails` set without image, `fetchImage` set to `true`, `avatarType` set to `photo` and contact specified via email | Contacts.Read | [/me/contacts/\*](/graph/api/user-list-contacts&tabs=cs) |
```
Comment (beth-panx) [inline concepts/toolkit/components/person.md:212]: broken link

File: concepts/toolkit/components/person.md [hunk 8]
```diff
@@ -204,15 +204,20 @@ The following properties are available on the config object.
...
+| `userId` set | User.ReadBasic.All | | [/users/{id}](/graph/api/user-list-people) |
```
Comment (beth-panx) [inline concepts/toolkit/components/person.md:213]: extra column
```suggestion
| `userId` set | User.ReadBasic.All | [/users/{id}](/graph/api/user-list-people) |
```

File: concepts/toolkit/components/person.md [hunk 9]
```diff
@@ -204,15 +204,20 @@ The following properties are available on the config object.
...
+| `personQuery` set to `me` and `avatarType` set to `photo` | User.Read | [/me/photo/$value](/graph/api/profilephoto-get) |
+| `personQuery` set to `me` and `avatarType` set to something else than `photo` | User.Read | [/me](/graph/api/user-get) |
+| `personQuery` set to a value other than `me` and `useContactApis` set to `true` | People.Read, User.ReadBasic.All, Contacts.Read | [/me/people/?$search=](/graph/api/user-list-people), [/users?$search=](/graph/api/user-list-people), [/me/contacts/\*](/graph/api/user-list-contacts&tabs=cs) |
```
Comment (beth-panx) [inline concepts/toolkit/components/person.md:216]: broken link for contact
also not sure if the /user?search= reference is correct, should it be /graph/api/user-list instead?

File: concepts/toolkit/components/person.md [hunk 10]
```diff
@@ -204,15 +204,20 @@ The following properties are available on the config object.
...
+| `personQuery` set to a value other than `me` and `useContactApis` set to `false` | People.Read, User.ReadBasic.All | [/me/people/?$search=](/graph/api/user-list-people), [/users?$search=](/graph/api/user-list-people) |
```
Comment (beth-panx) [inline concepts/toolkit/components/person.md:217]: not sure if the /user?search= reference is correct, should it be /graph/api/user-list instead?

File: concepts/toolkit/components/tasks.md [hunk 11]
```diff
@@ -149,12 +149,15 @@ The following example defines a template for the tasks component.
...
 
-| Resource | Permission |
-| - | - |
-| /me/planner/plans | Group.Read.All |
-| /planner/plans/${id} | Group.Read.All, Group.ReadWrite.All |
-| /planner/tasks | Group.ReadWrite.All |
-| /groups/${group-id}/planner/plans | Group.Read.All, Group.ReadWrite.All |
+| Configuration | Permission | API |
+| ------------- | ---------- | --- |
+| `groupId` set and `dataSource` set to `TasksSource.planner` | Group.Read.All | [/groups/${group-id}/planner/plans](/graph/api/plannergroup-list-plans?view=graph-rest-1.0&tabs=http), [/planner/plans/${planId}/buckets](/graph/api/plannerplan-list-buckets?view=graph-rest-1.0&tabs=http), [/planner/buckets/${bucketId}/tasks](graph/api/plannerplan-list-tasks?view=graph-rest-1.0&tabs=http) |
```
Comment (beth-panx) [inline concepts/toolkit/components/tasks.md:154]: last link is broken [/planner/buckets/${bucketId}/tasks](graph/api/plannerplan-list-tasks?view=graph-rest-1.0&tabs=http)

File: concepts/toolkit/components/tasks.md [hunk 12]
```diff
@@ -149,12 +149,15 @@ The following example defines a template for the tasks component.
...
+| `targetId` set and `dataSource` set to `TasksSource.todo` | Tasks.Read | [/me/outlook/taskGroups](/graph/api/outlookuser-list-taskgroups?view=graph-rest-beta&tabs=http&viewFallbackFrom=graph-rest-1.0), [/me/outlook/taskGroups/${groupId}/taskFolders](/graph/api/outlooktaskfolder-list-tasks?view=graph-rest-beta&tabs=http), [/me/outlook/taskFolders/${folderId}/tasks](/graph/api/outlooktaskfolder-list-tasks?view=graph-rest-beta&tabs=http) |
+| `targetId` set and `dataSource` set to something else than `TasksSource.todo` | Group.Read.All | [/planner/plans/${planId}](/graph/api/plannerplan-get?view=graph-rest-1.0&tabs=http), [/planner/plans/${planId}/buckets](/graph/api/plannerplan-list-buckets?view=graph-rest-1.0&tabs=http), [/planner/buckets/${bucketId}/tasks](graph/api/plannerplan-list-tasks?view=graph-rest-1.0&tabs=http) |
```
Comment (beth-panx) [inline concepts/toolkit/components/tasks.md:156]: same here, last link is broken [/planner/buckets/${bucketId}/tasks](graph/api/plannerplan-list-tasks?view=graph-rest-1.0&tabs=http)

File: concepts/toolkit/components/tasks.md [hunk 13]
```diff
@@ -149,12 +149,15 @@ The following example defines a template for the tasks component.
...
+| `dataSource` set to `TasksSource.planner` | Group.Read.All | [/me/planner/plans](/graph/api/planneruser-list-plans?view=graph-rest-1.0&tabs=http), [/planner/plans/${planId}/buckets](/graph/api/plannerplan-list-buckets?view=graph-rest-1.0&tabs=http), [/planner/buckets/${bucketId}/tasks](graph/api/plannerplan-list-tasks?view=graph-rest-1.0&tabs=http) |
```
Comment (beth-panx) [inline concepts/toolkit/components/tasks.md:157]: same here, last link is broken [/planner/buckets/${bucketId}/tasks](graph/api/plannerplan-list-tasks?view=graph-rest-1.0&tabs=http)

**Review by beth-panx — CHANGES_REQUESTED** [review]
Great work! Thanks for helping with the revamp of this. It's going to be so much easier for developers to understand the association of properties and permissions.
I pulled up preview and compared permissions to the docs link, found some differences and broken links.

**Comment by waldekmastykarz** [comment]
Thank you for such a thorough review @beth-panx and sorry for missing the broken links. Should've checked better.

File: concepts/toolkit/components/people-picker.md (hunk 1 above)
Comment (waldekmastykarz) [reply concepts/toolkit/components/people-picker.md:137]: Looking at the code, it seems like when you set `groupId`, we call `findGroupMembers` which requires the scopes I mentioned:

https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/87be56d08cfd3688c08a8c13841086e060f4899e/packages/mgt-components/src/components/mgt-people-picker/mgt-people-picker.ts#L789-L792

https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/87be56d08cfd3688c08a8c13841086e060f4899e/packages/mgt-components/src/graph/graph.user.ts#L322

File: concepts/toolkit/components/people-picker.md (hunk 2 above)
Comment (waldekmastykarz) [reply concepts/toolkit/components/people-picker.md:139]: Fixed

File: concepts/toolkit/components/people.md (hunk 3 above)
Comment (waldekmastykarz) [reply concepts/toolkit/components/people.md:100]: Adjusted.

File: concepts/toolkit/components/person.md (hunk 4 above)
Comment (waldekmastykarz) [reply concepts/toolkit/components/person.md:209]: Fixed

File: concepts/toolkit/components/person.md (hunk 5 above)
Comment (waldekmastykarz) [reply concepts/toolkit/components/person.md:210]: You're right. Adjusted

File: concepts/toolkit/components/person.md (hunk 6 above)
Comment (waldekmastykarz) [reply concepts/toolkit/components/person.md:211]: Fixed

File: concepts/toolkit/components/person.md (hunk 7 above)
Comment (waldekmastykarz) [reply concepts/toolkit/components/person.md:212]: Fixed

File: concepts/toolkit/components/person.md (hunk 8 above)
Comment (waldekmastykarz) [reply concepts/toolkit/components/person.md:213]: Removed

File: concepts/toolkit/components/person.md (hunk 9 above)
Comment (waldekmastykarz) [reply concepts/toolkit/components/person.md:216]: So this is coming from https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/87be56d08cfd3688c08a8c13841086e060f4899e/packages/mgt-components/src/components/mgt-person/mgt-person.ts#L986 where we call `findUsers` which does `/users?$search` https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/87be56d08cfd3688c08a8c13841086e060f4899e/packages/mgt-components/src/graph/graph.user.ts#L288-L291

Comment (waldekmastykarz) [reply concepts/toolkit/components/person.md:216]: Link for contact fixed

File: concepts/toolkit/components/person.md (hunk 10 above)
Comment (waldekmastykarz) [reply concepts/toolkit/components/person.md:217]: This is coming from https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/87be56d08cfd3688c08a8c13841086e060f4899e/packages/mgt-components/src/components/mgt-person/mgt-person.ts#L986 where we call findUsers which does /users?$search https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/87be56d08cfd3688c08a8c13841086e060f4899e/packages/mgt-components/src/graph/graph.user.ts#L288-L291

File: concepts/toolkit/components/tasks.md (hunk 11 above)
Comment (waldekmastykarz) [reply concepts/toolkit/components/tasks.md:154]: Fixed

File: concepts/toolkit/components/tasks.md (hunk 12 above)
Comment (waldekmastykarz) [reply concepts/toolkit/components/tasks.md:156]: Fixed

File: concepts/toolkit/components/tasks.md (hunk 13 above)
Comment (waldekmastykarz) [reply concepts/toolkit/components/tasks.md:157]: Fixed

**Comment by beth-panx** [comment]
> Thank you for such a thorough review @beth-panx and sorry for missing the broken links. Should've checked better.

Oh no worries! That's why we have reviews haha. Thanks for making this change!

**Review by beth-panx — APPROVED** [review]
🚀

**Pull request merged**